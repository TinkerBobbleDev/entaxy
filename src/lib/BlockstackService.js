const blockstack = require('blockstack')

window.blockstack = blockstack

export default blockstack
