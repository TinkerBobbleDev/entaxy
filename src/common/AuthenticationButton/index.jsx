import React from 'react'
import Button from 'material-ui/Button'
import { withStyles } from 'material-ui/styles'
import BlockstackService from './../../lib/BlockstackService'
import { getName } from '../../lib/IdentityService'

const styles = {}

function handleLogin() {
  BlockstackService.redirectToSignIn()
}

function handleLogout() {
  BlockstackService.signUserOut(window.location.href)
}

const AuthenticationButton = () => {
  if (BlockstackService.isUserSignedIn()) {
    const userData = BlockstackService.loadUserData()
    const { profile } = userData
    const person = new BlockstackService.Person(profile)

    return (
      <Button color="inherit" onClick={handleLogout}>
        {getName(person)} Logout
      </Button>
    )
  } else if (BlockstackService.isSignInPending()) {
    BlockstackService.handlePendingSignIn()
      .then(() => {
        window.location = window.location.origin
      })
  }

  return (<Button color="inherit" className="login" onClick={handleLogin}>Login</Button>)
}

AuthenticationButton.propTypes = {}

export default withStyles(styles)(AuthenticationButton)
