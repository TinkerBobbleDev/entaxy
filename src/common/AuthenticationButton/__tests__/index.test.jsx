import React from 'react'
import renderer from 'react-test-renderer'
import { mount, render } from 'enzyme'
import Button from 'material-ui/Button'
import AuthenticationButton from '../'
import BlockstackService from './../../../lib/BlockstackService'

jest.mock('./../../../lib/BlockstackService', () => {
  return require('../../../../mocks/BlockstackServiceMock')
})

beforeEach(() => {
  jest.resetModules()
  jest.clearAllMocks()
})

describe('AuthenticationButton', () => {
  it('matches snapshot', () => {
    const component = renderer.create(<AuthenticationButton />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  describe('logged in state', () => {
    it('should render the logout button', () => {
      const spy = jest.spyOn(BlockstackService, 'isUserSignedIn')
      spy.mockReturnValue(true)
      expect(render(<AuthenticationButton />).text()).toEqual('mocked name Logout')
      expect(spy).toHaveBeenCalled()
    })

    it('should render the name as nameless person if the name is not provided', () => {
      const spy = jest.spyOn(BlockstackService, 'isUserSignedIn')
      spy.mockReturnValue(true)

      BlockstackService.Person = jest.fn(() => {
        return {
          name: () => {
            return null
          }
        }
      })

      expect(render(<AuthenticationButton />).text()).toEqual('Nameless Person Logout')
      expect(spy).toHaveBeenCalled()
    })

    it('should call signUserOut when clicking the logout button', () => {
      const signUserOutSpy = jest.spyOn(BlockstackService, 'signUserOut')
      const isSignInPendingSpy = jest.spyOn(BlockstackService, 'isSignInPending')
      isSignInPendingSpy.mockReturnValue(false)
      const spy = jest.spyOn(BlockstackService, 'isUserSignedIn')
      spy.mockReturnValue(true)
      const wrapper = mount(<AuthenticationButton />)
      wrapper.find(Button).simulate('click')
      expect(signUserOutSpy.mock.calls.length).toBe(1)
    })
  })

  describe('logged out state', () => {
    it('should render the login button', () => {
      const spy = jest.spyOn(BlockstackService, 'isUserSignedIn')
      spy.mockReturnValue(false)
      expect(render(<AuthenticationButton />).text()).toEqual('Login')
      expect(spy).toHaveBeenCalled()
    })

    it('should call redirectToSignIn when clicking the login button', () => {
      const redirectToSignInSpy = jest.spyOn(BlockstackService, 'redirectToSignIn')
      const isSignInPendingSpy = jest.spyOn(BlockstackService, 'isSignInPending')
      isSignInPendingSpy.mockReturnValue(false)
      const wrapper = mount(<AuthenticationButton />)
      wrapper.find(Button).simulate('click')
      expect(redirectToSignInSpy.mock.calls.length).toBe(1)
    })
  })
})
