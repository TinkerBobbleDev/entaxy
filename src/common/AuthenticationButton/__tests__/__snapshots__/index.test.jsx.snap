// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticationButton matches snapshot 1`] = `
<button
  className="MuiButtonBase-root-16 MuiButton-root-1 MuiButton-colorInherit-5 login"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-2"
  >
    Login
  </span>
  <span
    className="MuiTouchRipple-root-18"
  />
</button>
`;
